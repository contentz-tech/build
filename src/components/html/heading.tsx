import { jsx } from "@emotion/core";
import slugify from "react-slugify";

const H1 = (props: any) =>
  jsx("h1", {
    css: {
      fontSize: "2.25em",
      fontWeight: "normal",
      letterSpacing: "-0.028em",
      margin: "1em 0"
    },
    id: slugify(jsx("h1", props)),
    ...props
  });

const H2 = (props: any) =>
  jsx("h2", {
    css: {
      fontSize: "2em",
      fontWeight: "normal",
      margin: "1em 0 0.25em"
    },
    id: slugify(jsx("h2", props)),
    ...props
  });

const H3 = (props: any) =>
  jsx("h3", {
    css: {
      fontSize: "1.75em",
      fontWeight: "normal",
      margin: "1em 0 0.25em"
    },
    id: slugify(jsx("h3", props)),
    ...props
  });

const H4 = (props: any) =>
  jsx("h4", {
    css: {
      fontSize: "1.5em",
      fontWeight: "normal",
      margin: "1em 0 0.25em"
    },
    id: slugify(jsx("h4", props)),
    ...props
  });

const H5 = (props: any) =>
  jsx("h5", {
    css: {
      fontSize: "1.25em",
      fontWeight: "normal",
      margin: "1em 0 0.25em"
    },
    id: slugify(jsx("h5", props)),
    ...props
  });

const H6 = (props: any) =>
  jsx("h6", {
    css: {
      fontSize: "1.125em",
      fontWeight: "normal",
      margin: "1em 0 0.25em"
    },
    id: slugify(jsx("h6", props)),
    ...props
  });

export { H1, H2, H3, H4, H5, H6 };
