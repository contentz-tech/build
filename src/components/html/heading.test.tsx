/** * @jest-environment jsdom */
import '@testing-library/jest-dom/extend-expect'
import { jsx } from "@emotion/core";
import { render } from "@testing-library/react";
import { H1, H2, H3, H4, H5, H6} from "./heading";

describe("Heading", () => {
  test("Level 1", () => {
    const { asFragment, getByText } = render(<H1>Heading 1</H1>);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Heading 1/)).toHaveTextContent("Heading 1");
  });

  test("Level 2", () => {
    const { asFragment, getByText } = render(<H2>Heading 2</H2>);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Heading 2/)).toHaveTextContent("Heading 2");
  });

  test("Level 3", () => {
    const { asFragment, getByText } = render(<H3>Heading 3</H3>);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Heading 3/)).toHaveTextContent("Heading 3");
  });

  test("Level 4", () => {
    const { asFragment, getByText } = render(<H4>Heading 4</H4>);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Heading 4/)).toHaveTextContent("Heading 4");
  });

  test("Level 5", () => {
    const { asFragment, getByText } = render(<H5>Heading 5</H5>);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Heading 5/)).toHaveTextContent("Heading 5");
  });

  test("Level 6", () => {
    const { asFragment, getByText } = render(<H6>Heading 6</H6>);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Heading 6/)).toHaveTextContent("Heading 6");
  });
});
