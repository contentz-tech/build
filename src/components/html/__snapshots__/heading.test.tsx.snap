// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Level 1 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 2.25em;
  font-weight: normal;
  -webkit-letter-spacing: -0.028em;
  -moz-letter-spacing: -0.028em;
  -ms-letter-spacing: -0.028em;
  letter-spacing: -0.028em;
  margin: 1em 0;
}

<h1
    class="emotion-0"
    id="heading-1"
  >
    Heading 1
  </h1>
</DocumentFragment>
`;

exports[`Heading Level 2 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 2em;
  font-weight: normal;
  margin: 1em 0 0.25em;
}

<h2
    class="emotion-0"
    id="heading-2"
  >
    Heading 2
  </h2>
</DocumentFragment>
`;

exports[`Heading Level 3 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 1.75em;
  font-weight: normal;
  margin: 1em 0 0.25em;
}

<h3
    class="emotion-0"
    id="heading-3"
  >
    Heading 3
  </h3>
</DocumentFragment>
`;

exports[`Heading Level 4 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 1.5em;
  font-weight: normal;
  margin: 1em 0 0.25em;
}

<h4
    class="emotion-0"
    id="heading-4"
  >
    Heading 4
  </h4>
</DocumentFragment>
`;

exports[`Heading Level 5 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 1.25em;
  font-weight: normal;
  margin: 1em 0 0.25em;
}

<h5
    class="emotion-0"
    id="heading-5"
  >
    Heading 5
  </h5>
</DocumentFragment>
`;

exports[`Heading Level 6 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 1.125em;
  font-weight: normal;
  margin: 1em 0 0.25em;
}

<h6
    class="emotion-0"
    id="heading-6"
  >
    Heading 6
  </h6>
</DocumentFragment>
`;
